src/Compilers/Core/Portable/Symbols/Attributes/CommonAttributeData.cs

```diff
@@ -161,14 +158,12 @@ internal T GetConstructorArgument<T>(int i, SpecialType specialType)
         /// For user defined attributes VB allows duplicate named arguments and uses the last value.
         /// Dev11 reports an error for pseudo-custom attributes when emitting metadata. We don't.
         /// </remarks>
-        [return: MaybeNull]
-        internal T DecodeNamedArgument<T>(string name, SpecialType specialType, T defaultValue = default)
+        internal T? DecodeNamedArgument<T>(string name, SpecialType specialType, T defaultValue = default)
```

Comment:
`T defaultValue = default` is expected to warn but doesn't due to a bug. It should be `T? defaultValue = default` instead. See #44341

Reply:
Agreed. I saw this but it seemed outside the scope of the change I was making.